tests/st/ipam/test_ipam.py

```diff
@@ -0,0 +1,255 @@
+                # This means the IP is not assigned
+                assert str(ip) not in workload_ips, \
+                    "ipam show says IP %s is not assigned when it is!" % ip
+
+        # Tidy up workloads
+        for workload, host in workloads:
+            self.delete_workload(host, workload)
+        # Delete new pool
+        self.hosts[0].calicoctl("delete -f newpool.yaml")
+
+    @parameterized.expand([
+        # Can't use boolean values here :(
+        "False",
+        "True",
+    ])
+    def test_pool_wrap(self, make_static_workload_str):
+        """
+        Repeatedly create and delete workloads until the system re-assigns an IP.
+        """
+        unittest.skip("remove me")
+        # Convert make_static_workload_str into a bool
+        make_static_workload = False
+        if make_static_workload_str == "True":
+            make_static_workload = True
+
+        ipv4_subnet = netaddr.IPNetwork("192.168.46.0/25")
+        new_pool = {'apiVersion': 'v1',
+                    'kind': 'ipPool',
+                    'metadata': {'cidr': str(ipv4_subnet.ipv4())},
+                    }
+        self.hosts[0].writefile("newpool.yaml", yaml.dump(new_pool))
+        self.hosts[0].calicoctl("create -f newpool.yaml")
+
+        host = self.hosts[0]
+        i = 0
+        if make_static_workload:
+            static_workload = host.create_workload("static",
+                                                   image="workload",
+                                                   network=self.network)
+            i += 1
+
+        new_workload = host.create_workload("wldw-%s" % i,
+                                            image="workload",
+                                            network=self.network)
+        assert netaddr.IPAddress(new_workload.ip) in ipv4_subnet
+        original_ip = new_workload.ip
+        while True:
+            self.delete_workload(host, new_workload)
+            i += 1
+            new_workload = host.create_workload("wldw-%s" % i,
+                                                image="workload",
+                                                network=self.network)
+            assert netaddr.IPAddress(new_workload.ip) in ipv4_subnet
+            if make_static_workload:
+                assert new_workload.ip != static_workload.ip, "IPAM assigned an IP which is " \
+                                                              "still in use!"
+
+            if new_workload.ip == original_ip:
+                # We assign pools to hosts in /26's - so 64 addresses.
+                poolsize = 64
+                # But if we're using one for a static workload, there will be one less
+                if make_static_workload:
+                    poolsize -= 1
+                assert i >= poolsize, "Original IP was re-assigned before entire host pool " \
+                                      "was cycled through.  Hit after %s times" % i
```

Comment:
Does the IPAM assignment algorithm guarantee that it waits as long as possible before re-use?  I think it might be best-effort, might be best to check with Rob, say.